### ISSUE TITLE
SQL injection exploits

### ISSUE THREAD

**Issue by kallisti-dev** [SupremoRTD/LoyaltyBot#6]
So I was looking through the source, and I didn't find anything that prevented various common SQL injection attacks. I tried inputting this into shane's twitch chat:

    !';DROP/**/TABLE/**/commands;SELECT'

I thought this would work, since MySQL interprets comments as whitespace characters, multiline statements are enabled in your db connection, and quotations are not escaped anywhere. But, for some reason, the intended effect was not achieved. 

Any idea why this exploit didn't work? My guess was that the database user that the bot logs into doesn't have drop permission. Regardless of this particular exploit not working, I still believe the bot is susceptible to SQL injection attacks, due to the lack of connection.escape calls on user inputs.

**Comment by SupremoRTD** [comment]
I believe this is because of the pg module that's being used. It has basic sql inject protection.

**Comment by kallisti-dev** [comment]
Right, but those mechanisms aren't being used. You have to call connection.escape manually to your inputs before you make the query. Alternatively, you can use the "?" syntax and the library will it do it automatically. None of the queries I've looked at have make use of either of those features.

**Comment by SupremoRTD** [comment]
I'll eventually take a look at this. More then likely it's a permission
issue.

I haven't taken a look at this code in years.
